### PR TITLE
Add: Test Mode badge to Classic Checkout, Pay for Order, and Add Payment Method

### LINKED ISSUE #9487: Applying V2 card styles to V1 element: Test mode flag (shortcode)
### Description

This is a follow-up issue to #9474. 

### Acceptance criteria

We would like to extend our V2 payment element designs to the V1 payment element. In #9474, we added the test mode flag to the blocks checkout. This should apply the same change to the shortcode checkout as well. 

> In Sandbox or in Test mode, the test copy is displayed.

### Designs

Please see #9474 and #9483 where this has been implemented on the blocks checkout for a visual comparison. 

### Dev notes

See #9483 

### Additional context

pfHfG4-gT-p2#comment-211

### PR DESCRIPTION
**Pull request by danielmx-dev**
Fixes #9487

#### Changes proposed in this Pull Request

* Add a "Test Mode" badge to Classic Checkout, Pay for Order, and Add Payment Method page that will match the badge already present in blocks checkout added by https://github.com/Automattic/woocommerce-payments/pull/9483
  * The styles are implemented using a grid layout to maintain consistency when the label size varies.

#### Testing Instructions - Prerequisites

* Enable test mode.
* Enable all payment methods.
* Enable the following currencies: USD, EUR
* Go to checkout

#### Testing Instructions - Checkout

* Add a product to the cart over 50 USD and go to classic checkout
* Verify that the test mode badge is displayed and that the layout is consistent whether the options are collapsed or expanded and in different resolutions.
  * Desktop
image
  * Mobile 
image
* Change the currency to EUR, verify that the payment methods are properly hidden
  * image
* Change the address country between Belgium, Netherlands, Poland and verify that payment methods dynamically show/hide
  * image 
* Temporarily disable test mode and validate that the layout is consistent, but the badge is no longer displayed
  * Desktop
image
  * Mobile
image
* Re-enable test mode

#### Testing Instructions - Add Payment Method

* Go to My Account > Payment Methods > Add Payment Method
* Validate that the test mode badge is displayed
  * image
* Temporarily disable test mode and validate that the layout is preserved Re-enable test mode afterwards.

#### Testing Instructions - Pay for order
 
* In WP Admin, create an order and open the customer payment page as a shopper
* Validate that the test mode badge is displayed
  * image
 
#### Testing Instructions - Regression
* In WP Admin, open a paid order.
* Validate that the payment method string ONLY includes the name, and no additional characters/HTML is rendered.
  * image

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [ ] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `9495` or branch name `add/test-mode-badge-classic-checkout` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&woocommerce-payments-dev-tools&branches.woocommerce-payments=add/test-mode-badge-classic-checkout) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: 64f0daabceeacbf21f3dedf0e84a649a60fcf8d1
- Build time: 2024-10-15 16:25:58 UTC

_Note: the build is updated when a new commit is pushed to this PR._

**Comment by danielmx-dev** [comment]
After reviewing my own screenshots I noticed that the badge looked a bit off, so I just pushed a commit to vertically center it.

image

Despite looking a bit off in the description screenshots, the badge should be properly centered now during testing.

File: includes/class-wc-payment-gateway-wcpay.php [hunk 1]
```diff
@@ -572,6 +572,26 @@ public function init_hooks() {
...
+	 * Returns the gateway title
+	 *
+	 * @return string
+	 * */
+	public function get_title() {
+		$title = parent::get_title();
+
+		if ( is_checkout() || is_add_payment_method_page() ) {
+			if ( WC_Payments::mode()->is_test() ) {
+				$test_mode_badge = '<span class="test-mode badge">Test Mode</span>';
```
Comment (frosso) [inline includes/class-wc-payment-gateway-wcpay.php]: We'll need to use the `__()` i18n wrapper, here

**Review by frosso — DISMISSED** [review]
Some small changes for i18n. I'll take care of them tomorrow.

I also wonder if we could benefit from some horizontal spacing between the label and the payment method name
Screenshot 2024-09-30 at 4 11 03 PM

**Comment by frosso** [comment]
Screenshot 2024-10-01 at 6 16 31 PM
Updated to have i18n wrapper, added 8px spacing in 7b273a6e4 .

Re-assigning for internal review, as discussed in team meeting.

**Comment by gpressutto5** [comment]
@danielmx-dev @frosso FYI paJDYF-foh-p2#comment-26617

**Comment by danielmx-dev** [comment]
I have pushed a commit https://github.com/Automattic/woocommerce-payments/pull/9495/commits/4daaf36fb87f854ee50673374ec75f2fea770443 to only render the badge for the credit card payment method as it was already done for blocks checkout in https://github.com/Automattic/woocommerce-payments/pull/9514

image

**Review by gpressutto5 — CHANGES_REQUESTED** [review]
| image | image |
|-----|------|

Can we move the label to the text line when there's not enough space instead of changing the logo size? Like we do in the blocks checkout:

image

**Comment by danielmx-dev** [comment]
I pushed an update to push the badge if there is not horizontal space.

image

If there's enough space the badge will appear next to the payment method title
image

**Comment by danielmx-dev** [comment]
There are some themes that do not hide the radio button like Storefront does (e.g. twenty twenty-four), which was causing issues with the layout. I have updated the styles to make sure the styles look good in that scenario.

image

Since the payment methods other than card won't use the badge, I won't apply any of the new complex styles on them, to keep things simpler.

**Review by gpressutto5 — APPROVED** [review]
It looks great!

image

**Comment by danielmx-dev** [comment]
The alignment of the title was a bit off so I added a couple of commits to fix it. It should be fine now.

Storefront: 
image
image

Twenty twenty-four
image

Twenty twenty-two
image

**Pull request merged**